---
title: "R Notebook"
output: html_notebook
---
```{r}
library(here)
library(tidyverse)

here::here()
data_dir <- here("ivigs/")
raw_dir <- here("ivigs/")

pal2 <- c("#015b58", "#5962b5")
pal2.flip <- c("#5962b5", "#015b58")
```

```{r}
#box <- read.delim2(here(raw_dir, "0A_alldan_consensus_py.txt"))
#box <- subset(box, box$Mismatches==0 | box$Mismatches==1)
#write.csv(box, here(raw_dir, "perf1mis_0a_alldan_consensus_py.txt"))

box <- read.csv(here(raw_dir, "only_perfect0a_alldan_consensus_py.txt"), row.names = 1)
#box <- read.csv(here(raw_dir, "perf1mis_0a_alldan_consensus_py.txt"), row.names = 1)
#box <- subset(box, box$Mismatches==1)

box <- box %>% rename(product = Sequence, phage = Contig)
box[,c(2)] <- "0A_box"
box <- box[,c(3,2,1)]
box$count <- 1

dan.tax <- read.csv(here(raw_dir, "dan_fam_spor.csv" ), row.names=1) 
#dan.tax <- dan.tax[,c(2,1,3,4,29)]

#phatyp <- read.delim2(here(raw_dir, "dan_phatyp_prediction.tsv" )) 
style <- read.delim2(here(raw_dir, "dan_phastyle_predictions.tsv" )) 

checkv <- read.delim2(here("ivigs/phagescope_metadata.tsv" )) 
checkv <- checkv[,c(1,5)]


all <- merge(box, style, by.x="phage", by.y="fasta_id", all.x=TRUE, all.y=TRUE) 

all <- merge(all, dan.tax, by.x="phage", by.y="ID", all.x=TRUE, all.y=FALSE)
all$life_bool <- ifelse(all$predicted_label=="temperate", TRUE, FALSE)

all <- merge(all, checkv, by.x="phage", by.y="Phage_ID", all.x=TRUE, all.y=FALSE)

all <- subset(all, all$f_spor==TRUE)
all <- subset(all, all$Completeness=="Complete" | all$Completeness=="High-quality" ) #| all$Completeness=="Medium-quality")

all$count[is.na(all$count)] <- 0

all %>% group_by(predicted_label, count) %>% summarise(n=n())

df <- data.frame("w0a" = c(36889, 9687), "n0a" = c(243, 54), row.names = c("temperate", "lytic"))  ## complete, high

#df <- data.frame("w0a" = c(56715, 13297), "n0a" = c(707, 183), row.names = c("temperate", "lytic"))  ## all but low

#df <- data.frame("w0a" = c(68418, 16084), "n0a" = c(1961, 665), row.names = c("temperate", "lytic"))  ## all 


fisher.test(df)  ## NO. 




```


```{r}
group <- all %>% group_by(phage, predicted_label, gtdb_f) %>% summarise(n=n()) 

group$count <- 1

mean <- all %>% group_by(phage, predicted_label ) %>% summarise(n=n()) 

all %>% group_by(predicted_label ) %>% summarise(n=n()) 

host <- group %>% group_by(predicted_label, gtdb_f) %>% summarise(count=n()) 


group %>% group_by(predicted_label) %>% summarise(n=mean(n)) 




ggplot(group, aes(x = factor(predicted_label), fill = factor(predicted_label), y = n)) +
  geom_boxplot(binaxis = "y", stackdir = "center", position = "dodge") + geom_jitter(width = 0.2) + scale_fill_manual(values=pal2, labels = c ("Temperate", "Lytic")) + ylab("# of 0A box / phage genome") + xlab("Phage Lifestyle") + ggtitle("0A Box Count by Phage Lifestyle (GPD/GVD)") + labs(fill="Lifestyle")

 ggsave(here("ivigs/0Abox_grouped_all.png"), height =4 ,width = 6) 

ggplot(group, aes(x = factor(gtdb_f), fill = factor(predicted_label), y = n)) +
  geom_boxplot(binaxis = "y", stackdir = "center", position = "dodge") +coord_flip() + scale_fill_manual(values=pal2, labels = c ("Temperate", "Lytic"))  + ylab("# of 0A box / phage genome") + xlab("Host Taxa") + ggtitle("0A Box Count by Phage Lifestyle (GPD/GVD)") + labs(fill="Lifestyle") #+ geom_jitter(width = 0.2) + coord_flip()

 ggsave(here("ivigs/0Abox_taxa_all.png"), height =4 ,width = 6) 

ggplot(group, aes(x = factor(gtdb_f), fill = factor(predicted_label), y = n)) +
  geom_boxplot(binaxis = "y", stackdir = "center", position = "dodge") + geom_jitter(width = 0.2) + coord_flip()+ scale_fill_manual(values=pal2, labels = c ("Temperate", "Lytic")) + ylab("# of 0A box / phage genome") + xlab("Host Taxa") + ggtitle("0A Box Count by Phage Lifestyle (GPD/GVD)") + labs(fill="Lifestyle")

bacillus <- subset(group, group$gtdb_f=="Bacillaceae" | group$gtdb_f=="Clostridiaceae" | group$gtdb_f=="Erysipelotrichaceae")

ggplot(bacillus, aes(x = factor(gtdb_f), fill = factor(predicted_label), y = n)) +
  geom_boxplot(binaxis = "y", stackdir = "center", position = "dodge")  + scale_fill_manual(values=pal2, labels = c ("Temperate", "Lytic"))   + ylab("# of 0A box / phage genome") + xlab("Phage Lifestyle") + ggtitle("0A Box Count by Phage Lifestyle (GPD/GVD)") + labs(fill="Lifestyle") 
#+ geom_jitter(position=position_dodge(0.8))

 ggsave(here("ivigs/0Abox_exampletaxa_all.png"), height =4 ,width = 6) 



```


```{r}

res.aov1 <- aov(n ~ predicted_label, data = group)

res.aov2 <- aov(n ~ predicted_label+gtdb_f, data = group)

res.aov3 <- aov(n ~ predicted_label*gtdb_f, data = group)

summary(res.aov3)

anova(res.aov1, res.aov2, res.aov3)

```

```{r}
ggplot
```

