---
title: "R Notebook"
output: html_notebook
---
```{r}
library(here)
library(tidyverse)

here::here()
data_dir <- here("refseq_0403/raw_data/")
raw_dir <- here("refseq_0403/raw_data/")

pal2 <- c("#015b58", "#5962b5")
pal2.flip <- c("#5962b5", "#015b58")
```

```{r}
#box <- read.delim2(here(raw_dir, "0A_inphjan25_knownsporphage0A_consensus_py.txt"))
#box <- subset(box, box$Mismatches==0 ) #| box$Mismatches==1)


#box <- box %>% rename(product = Sequence, phage = Contig)
#write.csv(box, here(raw_dir, "perf_0a_refseq_consensus_py.txt"))
box <- read.csv(here(raw_dir, "perf_0a_refseq_consensus_py.txt"), row.names=1)

spbc2.0A <- subset(box, box$phage=="AF020713")

box[,c(2)] <- "0A_box"
box <- box[,c(3,2,1)]
box$count <- 1

dan.tax <- read.csv(here(raw_dir, "known_sporhostphage.csv" ), row.names=1) 
dan.tax <- dan.tax[,c(2,1,3,4,29)]

phatyp <- read.delim2(here(raw_dir, "phabox_final_prediction_summary.tsv" )) 
style <- read.delim2(here("inphared/raw_data/inphared_phastyle_predictions.tsv" )) 



all <- merge(box, style, by.x="phage", by.y="fasta_id", all.x=TRUE, all.y=FALSE) 

all <- merge(all, dan.tax, by.x="phage", by.y="Accession", all.x=TRUE, all.y=FALSE)
all$life_bool <- ifelse(all$predicted_label=="temperate", TRUE, TRUE)
all <- subset(all, all$f_spor==TRUE)


all$count[is.na(all$count)] <- 0

phage.count <- all %>% group_by(predicted_label, count, phage, Description) %>% summarise(n=n())

group <- all %>% group_by(phage, predicted_label, Host) %>% summarise(n=n()) 

group$count <- 1

mean <- all %>% group_by(phage, predicted_label ) %>% summarise(n=n()) 

host <- group %>% group_by(predicted_label) %>% summarise(count=n()) 



df <- data.frame("w0a" = c(2583, 5733), "n0a" = c(43, 60), row.names = c("temperate", "lytic"))
df


fisher.test(df)  ## 0.02 

ggplot(group, aes(x = factor(predicted_label), fill = factor(predicted_label), y = n)) +
  geom_boxplot(binaxis = "y", stackdir = "center", position = "dodge") + geom_jitter(width = 0.2) + scale_fill_manual(values=pal2, labels = c ("Temperate", "Lytic")) + ylab("# of 0A box / phage genome") + xlab("Phage Lifestyle") + ggtitle("0A Box Count by Phage Lifestyle (INPHARED)") + labs(fill="Lifestyle")

 ggsave(here("refseq/out-dir/0A_group_inph.png"), height =4 ,width = 6)

ggplot(group, aes(x = factor(Host), fill = factor(predicted_label), y = n)) +
  geom_boxplot(binaxis = "y", stackdir = "center", position = "dodge") + coord_flip() + scale_fill_manual(values=pal2, labels = c ("Temperate", "Lytic")) + ylab("# of 0A box / phage genome") + xlab("Host Taxa") + ggtitle("0A Box Count by Phage Lifestyle (INPHARED)") + labs(fill="Lifestyle")


 ggsave(here("refseq/out-dir/0A_taxa_inph.png"), height =4 ,width = 6)

ggplot(group, aes(x = factor(Host), fill = factor(predicted_label), y = n)) +
  geom_boxplot(binaxis = "y", stackdir = "center", position = "dodge") + geom_jitter(width = 0.2) + coord_flip() + scale_fill_manual(values=pal2, labels = c ("Temperate", "Lytic")) + ylab("# of 0A box / phage genome") + xlab("Host Taxa") + ggtitle("0A Box Count by Phage Lifestyle (INPHARED)") + labs(fill="Lifestyle")


bacillus <- subset(group, group$Host=="Bacillus")

ggplot(bacillus, aes(x = factor(Host), fill = factor(predicted_label), y = n)) +
  geom_boxplot(binaxis = "y", stackdir = "center", position = "dodge") + scale_fill_manual(values=pal2, labels = c ("Temperate", "Lytic")) + ylab("# of 0A box / phage genome") + xlab("Phage Lifestyle") + ggtitle("0A Box Count by Phage Lifestyle (INPHARED)") + labs(fill="Lifestyle")
#+ geom_jitter(position=position_dodge(0.8))





```


```{r}

res.aov1 <- aov(n ~ predicted_label, data = group)

res.aov2 <- aov(n ~ predicted_label+Host, data = group)

res.aov3 <- aov(n ~ predicted_label*Host, data = group)




summary(res.aov3)

anova(res.aov1, res.aov2, res.aov3)

```

```{r}
ggplot
```

