---
title: "R Notebook"
output: html_notebook
---
```{r}
library(here)
library(tidyverse)

here::here()
data_dir <- here("refseq_0403/raw_data/")
raw_dir <- here("refseq_0403/raw_data/")


```

```{r}

### enriched sporulation genes from Dan paer
enrich2 <- read.csv(here(raw_dir, "viral_sporGenes.csv" )) 
hit <- enrich[,c(1,2)]

enrich <- read.delim2(here(raw_dir, "MASTERLIST_KEGGHIT_metacerb_final_annotation_summary.tsv" )) 
hit <- enrich[,c(3,8)]
hit$sporulation <- "sporulation_hit"
hit <- unique(hit)
hit <- subset(hit, hit$best_hit!="")


## annotation of refseq phages & flagging of spor genes.
#metacerb.best <- read.delim2(here(raw_dir, "22825_metacerb_all_fitered_annotation_hits.tsv" )) 
metacerb.kegg <- read.delim2(here(raw_dir, "metacerb_annotation_summary_KOFam_all_KEGG.tsv" )) ##d_amg
metacerb.kegg <- merge(metacerb.kegg, hit, by.x="best_hit", by.y="best_hit")
metacerb.kegg$gene_name[is.na(metacerb.kegg$gene_name)] <- ""
metacerb.kegg$sporulation[is.na(metacerb.kegg$sporulation)] <- "no_hit"
metacerb.best  <- metacerb.kegg
metacerb.best$phage<- gsub('_[^_]*$', '', metacerb.best$target) 

metacerb.best <- metacerb.best[,c(1,3,7,14,16)]


```

```{r}
### dan sporulation T/F information
dan.tax <- read.csv(here(raw_dir, "known_sporhostphage.csv" ), row.names=1) 
dan.tax <- dan.tax[,c(2,1,3,4,29)]


##phage lifestyle information
### some inphared phages missing
phatyp <- read.delim2(here(raw_dir, "phabox_final_prediction_summary.tsv" ))   # d_vir
#phagescope <- subset(phagescope, phagescope$Phage_source=="RefSeq" | phagescope$Phage_source=="GenBank")
#phagescope$phage<- gsub('.[^.]*$', '', phagescope$Phage_ID) 

```


```{r}
box <- read.delim2(here(raw_dir, "0A_inphjan25_knownsporphage0A_consensus_py.txt"))
box <- subset(box, box$Mismatches==0)

box <- box %>% rename(product = Header, gene = Sequence, gene_name = Position,  phage = Contig, best_hit = Mismatches)
box[,c(1,2,4,5)] <- "0A_box"

meta0a <- bind_rows(box, metacerb.best)

```


```{r}
### merge annotation data w/ phagescope lifestyle + sporulation status of bac host
#all <- merge(meta0a, phatyp, by.x="phage", by.y="Accession", all.x=TRUE, all.y=FALSE) 

all <- merge(metacerb.best, phatyp, by.x="phage", by.y="Accession", all.x=TRUE, all.y=TRUE) 

all <- merge(all, dan.tax, by.x="phage", by.y="Accession", all.x=TRUE, all.y=FALSE)
all$life_bool <- ifelse(all$TYPE=="temperate", TRUE, FALSE)
all <- subset(all, all$f_spor==TRUE)

```

```{r}
# add host sporulation to amg
d.amg <- all %>% rename(spor_host = f_spor, gene_id = gene_name, gene_description = product, Lifestyle = TYPE, Bac.Host = Host.y )

d.amg %>% group_by(Lifestyle) %>% summarise(n=n())
```


First we summarize the number of phages infecting hosts that sporulate or not
```{r}
d.vir <- d.amg %>% group_by(phage, spor_host, life_bool, Bac.Host, Lifestyle) %>% summarise(n=n())

```

```{r}
p.signif <- 1e-3

k_treshold <- 8
```



First we summarize the number of phages infecting hosts that sporulate or not
```{r}
d.host <- 
  d.vir %>% 
  group_by(life_bool) %>% 
  summarise(n=n())

total_temp <- 
  d.host %>% 
  filter(life_bool) %>% 
  pull(n) 

total_vir <- 
  d.host %>% 
  filter(!life_bool) %>% 
  pull(n) 

d.host
```


For each gene we ask how many times it is found in a phage infecting a sporulator or a non-sporulator host.
```{r}

# Summarise total number of sporulators and non sporulators
d.amg.sum <- d.amg %>%
  # filter(str_detect(header, regex("sporulation", ignore_case = T))) %>%
  filter(!is.na(gene_id)) %>% 
  filter(!is.na(life_bool)) %>% 
  group_by(gene_id, gene_description, life_bool) %>% #category removed, lifestyle added
  summarise(n = n()) %>%
  arrange(desc(n))

d.amg.sum %>%
  ggplot(aes(gene_id, n, color = life_bool))+
  geom_point(shape=21, size=2)+
  #facet_wrap(~spor_host, scales = "free_y")+ ## category removed, lifestyle added
  coord_flip()+
  theme_bw()



d.amg.sum$rel <- ifelse(d.amg.sum$life_bool==TRUE, d.amg.sum$n/total_temp*100, d.amg.sum$n/total_vir*100)


rel <- d.amg.sum %>%
  ggplot(aes(gene_id, rel, color = life_bool))+
  geom_point(shape=21, size=2)+
  ylab( "normalized (%) occurence from AMG genes") +
  coord_flip()+ theme_bw() + labs(color="Lifestyle") + scale_color_hue(labels = c("Virulent", "Temperate")) 
  


rel

# ggsave(here("refseq/out-dir/rel_lifestyle_spor_enrich.png"), height =3 ,width = 6)


```



```{r}
temp_enrich <- 
  d.amg %>% 
  
  # Summarize KOs by host sporulation
 mutate(life_bool = case_when(
    life_bool ~ "temperate",
    !life_bool ~ "virulent"
  )) %>% 
  filter(!is.na(gene_id)) %>% 
  group_by(gene_id, life_bool) %>% 
  summarise(n=n(), .groups = "drop") %>% 
  pivot_wider(values_from = n, names_from = life_bool, values_fill = 0) %>% 
   
    # q	: the number of white balls drawn without replacement 
    # from an urn which contains both black and white balls.
    # >>> AMG (specific gene) in viruses infecting sporulators
    mutate(q = temperate) %>% 
    
    # m	:the number of white balls in the urn.
    # >>> number of viruses infecting sporulators in the pool
    mutate(m = total_temp) %>% 
    
    # n	:the number of black balls in the urn.
    # >>> number of viruses infecting non-sporulators in the pool
    mutate(n = total_vir) %>% 
    
    # k: the number of balls drawn from the urn, hence must be in 0,1,., m+n.
    # >>> Total AMG (specific gene) detected in viruses
    mutate(k = virulent + temperate) %>% 
    
    mutate(p.val = signif (phyper(q, m, n, k, lower.tail =F),5)) #%>% 
  # Adjust Pvalue for multiple testing
 #   mutate(p.adj = signif (p.adjust(p.val, method = "BH"),5)) %>% 
  # mark significantly enriched genes
  # mutate(significant = (p.adj < p.signif) & (k>k_treshold))
#pval adjust done later
  
temp_enrich$enriched <- "temperate"
p_temp.all <- temp_enrich
```


```{r}
vir_enrich <- 
  d.amg %>% 
  
  # Summarize KOs by host sporulation
 mutate(life_bool = case_when(
    life_bool ~ "temperate",
    !life_bool ~ "virulent"
  )) %>% 
  filter(!is.na(gene_id)) %>% 
  group_by(gene_id, life_bool) %>% 
  summarise(n=n(), .groups = "drop") %>% 
  pivot_wider(values_from = n, names_from = life_bool, values_fill = 0) %>% 
   
    # q	: the number of white balls drawn without replacement 
    # from an urn which contains both black and white balls.
    # >>> AMG (specific gene) in viruses infecting that lifestyle
    mutate(q = virulent) %>% 
    
    # m	:the number of white balls in the urn.
    # >>> number of viruses of that lifestyle
    mutate(m = total_vir) %>% 
    
    # n	:the number of black balls in the urn.
    # >>> number of viruses infecting the other lifestyle
    mutate(n = total_temp) %>% 
    
    # k: the number of balls drawn from the urn, hence must be in 0,1,., m+n.
    # >>> Total AMG (specific gene) across all lifestyles
    mutate(k = virulent + temperate) %>% 
    
    mutate(p.val = signif (phyper(q, m, n, k, lower.tail =F),5)) #%>% 
  # Adjust Pvalue for multiple testing
    #mutate(p.adj = signif (p.adjust(p.val, method = "BH"),5)) %>% 
  # mark significantly enriched genes
   #mutate(significant = (p.adj < p.signif) & (k>k_treshold))

#pval adjust done later

vir_enrich$enriched <- "virulent"
p_vir.all <- vir_enrich

hfl <- function(q, m, n, k) {
  exp(lchoose(m,q) + lchoose(n, k-q) - lchoose(m+n, k))
}

hfl(2583, 459, 580, 8316)
```

```{r}

## pval adjustement for all?

all.pv <- rbind(p_temp.all, p_vir.all)
# all.pv$og.adj <- all.pv$p.adj


## i adjust p.val original, NOT p.adj (from original?)
  # Adjust Pvalue for multiple testing
all.pv.adj <-  all.pv %>%  mutate(p.adj = signif (p.adjust(p.val, method = "BH"),5)) %>% 
  # mark significantly enriched genes
   mutate(significant = (p.adj < p.signif) & (k>k_treshold))

##parA is **mostly** in temperate clostridium difficile phage -- think this might be a sampling issue.
```



Plot enrichment results

```{r}
# labels for significant genes
labels <- 
  d.amg %>% 
  select(gene_id,gene) %>% 
  distinct() %>% 
  filter(gene_id %in% all.pv.adj$gene_id[d.vir$significant]) %>% 
  mutate(lab = str_remove(gene, "[;|\\[].*")) %>% 
  group_by(gene_id) %>% 
  summarise(p_lab = str_c(lab, collapse = "\n")) %>% 
  #adjustment of one label
  mutate(p_lab = if_else(gene_id == "K00558", paste0(p_lab, "\n"), p_lab))

d_enrich <-
  left_join(all.pv.adj, labels,  by = "gene_id") 

d_enrich$real <- ifelse(d_enrich$significant==TRUE, d_enrich$gene_id, NA)


  #mark sporulation genes
d_enrich <- d_enrich %>% 
  mutate(spor_gene = if_else(gene_id %in% unique(enrich$gene_name),
         "Spoulation genes", "Other genes"))

# plot
p <-
  d_enrich %>% 
  filter(k>0) %>%
  ggplot(aes(k,-log10(p.adj+1e-120)))+
  geom_hline(yintercept = -log10(p.signif), linetype=2, color="grey", size = 1)+
  geom_vline(xintercept = k_treshold, linetype=2, color="grey", size = 1)+
  
  geom_text_repel(aes(label = real), size = 3, na.rm = T, point.padding = 0, min.segment.length = 0,seed = 123)+
  
  geom_jitter(aes(fill = significant),width = 0.05, height = 0.05,
              shape=21, size=3, stroke = 1, show.legend = F)+
  scale_fill_viridis_d(direction = -1, alpha = 0.5)+
  theme_classic()+
  facet_wrap(~ enriched %>% fct_rev(), ncol = 2) +
  panel_border(color = "black")+
  scale_x_continuous(trans = "log2", breaks = (2^(0:11)))+
  xlab("Sample size\nNo. homologs detected (log2)")+
  ylab("Enrichment (-log10 adj. P-value)")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1),
        strip.background = element_blank(),
        strip.text = element_text(face = "bold"))

# ggsave(here("refseq/out-dir/refseq_lifestyle_spor_enrich.png"), height =4 ,width = 6)
p

```