---
title: "R Notebook"
output: html_notebook
---

```{r}
library(here)
library(tidyverse)
pal2 <- c("#015b58", "#5962b5")
pal2.flip <- c("#5962b5", "#015b58")
here::here()
data_dir <- here("inphared/")
raw_dir <- here("inphared/raw_data/")

#box <- read.delim2(here(raw_dir, "0A_inphjan25_consensus_py.txt"))
#box <- subset(box, box$Mismatches==0)
#write.csv(box, here(raw_dir, "perf0A_inphjan25_consensus_py.txt"))

box <- read.csv(here(raw_dir, "perf0A_inphjan25_consensus_py.txt"), row.names = 1)
#box <- read.csv(here(raw_dir, "perf1mis_0a_alldan_consensus_py.txt"), row.names = 1)
#box <- subset(box, box$Mismatches==1)

box <- box %>% rename(product = Sequence, phage = Contig)
box[,c(2)] <- "0A_box"
box <- box[,c(3,2,1)]
box$count <- 1

box <-  box %>% group_by(phage, product) %>% summarise(sum=sum(count)) 

dan.tax <- read.csv(here("isolate_dbs/gtdb/gtdb_families_sporulation.csv" ))
dan.tax$gtdb_f<- gsub('f__', '', dan.tax$gtdb_f) 
dan.tax <- dan.tax[,c(5,6)]
dan.tax <- unique(dan.tax)

gtdb.tax <- read.csv(here("isolate_dbs/gtdb/gtdb-taxonomy-table.csv" ))
gtdb.tax <- subset(gtdb.tax, gtdb.tax$Domain=="Bacteria")
colnames(gtdb.tax) <- paste0("newgtdb_", colnames(gtdb.tax))

gtdb.tax <- merge(dan.tax, gtdb.tax, by.x="gtdb_f", by.y="newgtdb_Family", all.x=TRUE, all.y=TRUE)

gtdb.tax <- gtdb.tax[,c(7,2)]
gtdb.tax <- unique(gtdb.tax)

inph <- read.delim2(here(raw_dir, "1Jan2025_data.tsv"))




#phatyp <- read.delim2(here(raw_dir, "dan_phatyp_prediction.tsv" )) 
style <- read.delim2(here(raw_dir, "inphared_phastyle_predictions.tsv" )) 

all <- merge(box, style, by.x="phage", by.y="fasta_id", all.x=TRUE, all.y=TRUE) 
```

box$count <- 1

box <-  box %>% group_by(phage, product, Host, predicted_label, f_spor) %>% summarise(sum=sum(count)) 
all <- box

```{r}

inph <- read.delim2(here("isolate_dbs/inphared/1Jan2025_data_excluding_refseq.tsv"))
inph2 <- read.csv(here("isolate_dbs/inphared/known_sporhostphage.csv"))
gtdb.tax <- read.csv(here("isolate_dbs/gtdb/gtdb-taxonomy-table.csv" ))
gtdb.tax <- subset(gtdb.tax, gtdb.tax$Domain=="Bacteria")
colnames(gtdb.tax) <- paste0("newgtdb_", colnames(gtdb.tax))

#gtdb.spor <- read.csv(here("isolate_dbs/gtdb/gtdb_families_sporulation.csv" ))

meta <- merge(inph, gtdb.tax, by.x="Host", by.y="newgtdb_Genus", all.x=TRUE, all.y=FALSE)

nonbacil <- meta[,c(2,29)]
nonbacil$bacill <- 1
nonbacil$bacill <- ifelse(nonbacil$newgtdb_Phylum=="Bacillota", TRUE, FALSE)
colnames(nonbacil) <- c("phage", "phyla", "bacilli")
nonbacil <-  subset(nonbacil, !grepl("NC_", phage))
nonbacil <- unique(nonbacil)

box <- read.csv(here(raw_dir, "perf0A_inphjan25_consensus_py.txt"), row.names = 1)
#box <- read.csv(here(raw_dir, "perf1mis_0a_alldan_consensus_py.txt"), row.names = 1)
#box <- subset(box, box$Mismatches==1)

box <- box %>% rename(product = Sequence, phage = Contig)
box[,c(2)] <- "0A_box"
box <- box[,c(3,2,1)]
box$count <- 1

box2 <-  box %>% group_by(phage, product) %>% summarise(sum=sum(count)) 
box2 <-  subset(box2, !grepl("NC_", phage))


all <- merge(box2, nonbacil, by.x="phage", by.y="phage", all.x=TRUE, all.y=TRUE)
all <-  subset(all, !grepl("NC_", phage))


all$catspor <- "0"

all$product[is.na(all$product)] <- "0A_box"

all$bacilli[is.na(all$bacilli)] <- "unknown"
all <- subset(all, all$bacilli!="unknown")

all$sum[is.na(all$sum)] <- 0

all <- merge(all, inph2, by.x="phage", by.y="Accession", all.x=TRUE, all.y=FALSE)



all$f_spor[is.na(all$f_spor)] <- "unknown"




#all2$life_bool <- ifelse(all2$predicted_label=="temperate", TRUE, FALSE)

all$f_spor[is.na(all$f_spor)] <- "unknown"
all$f_spor <- as.character(all$f_spor)


all$sporulation <- "OtherPhyla"
all$sporulation <- ifelse(all$f_spor=="TRUE", "Spor", all$sporulation)
all$sporulation <- ifelse(all$f_spor=="FALSE", "NonSpor", all$sporulation)
all$sporulation<-  ifelse(all$bacilli==FALSE, "OtherPhyla", all$sporulation)


all$label <- "OtherPhyla"
all$label <- ifelse(all$f_spor==TRUE, "Bacill_Spor", all$label)
all$label <- ifelse(all$f_spor==FALSE, "Bacill_NonSpor", all$label)

#all <- subset(all, all$f_spor!="unknown")

#all <- unite(all, "phage_type", c("sporulation", "predicted_label"), sep = "_", remove = FALSE, na.rm = FALSE)


all <-  subset(all, !grepl("NC_", phage))

all.deet <- all

all$sporulation[is.na(all$sporulation)] <- "unknown"
all <- subset(all, all$sporulation!="unknown")

all <- all[,c(1:5,41,42,43)]


all$tally <- 1
all$hit <- ifelse(all$sum!=0, 1, 0)

counts <- all %>% group_by(sporulation, bacilli) %>% summarise(num.phage=sum(tally), total.0A=sum(sum), mean.0A=mean(sum), hits.0A=sum(hit)) 




## 

counts$perc.hit <- counts$hits.0A/counts$num.phage


counts


```





```{r}

spor.stat <- c('Spore', 'Non-spore')
outcome <- c('Hits', 'None')
data <- matrix(c(811, 75, 2234, 298), nrow=2, ncol=2, byrow=TRUE)
dimnames(data) <- list('Spore'=spor.stat, 'Outcome'=outcome)


library(epitools)

fisher.test(data)  ## 


ggplot(all, aes(x = factor(sporulation), fill = factor(sporulation), y = sum )) +
  geom_boxplot(binaxis = "y", stackdir = "center", position = "dodge") + geom_jitter(width = 0.2) + ylab("Number of 0A boxes")


ggplot(all, aes(Genome.Length..bp.,sum, fill=phage_type, colour = phage_type)) +
  geom_point() + ylab("Number of 0A boxes per genome") + xlab("Phage Genome (bp)") +
  geom_smooth(method="lm")+ # Add regression line
 stat_regline_equation(aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~~")),
                        label.x.npc = "middle", label.y.npc = "top" )# Add equation#+ ylim(0, 25) #+ xlim(0,




ggplot(all, aes(x = factor(phage_type), fill = factor(phage_type), y = sum )) +
  geom_boxplot(binaxis = "y", stackdir = "center", position = "dodge") + geom_jitter(width = 0.2) + ylab("Number of 0A boxes")



p <- ggplot(all, aes(x=phage_type, y=sum, fill=phage_type)) +
  geom_violin(trim=TRUE)+
  geom_boxplot(width=0.1) 

p

p+ geom_jitter(binaxis='y', stackdir='center', dotsize=0.001, shape=1)

p+ geom_dotplot(binaxis='y', stackdir='center', dotsize=0.05, shape=1)

```


```{r}


library(ggpubr)

ggplot(all.deet, aes(Genome.Length..bp.,sum, fill=phage_type, colour = phage_type)) +
  geom_point() + ylab("Number of 0A boxes per genome") + xlab("Phage Genome (bp)") +
  geom_smooth(method="lm")+ # Add regression line
 stat_regline_equation(aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~~")),
                        label.x.npc = "middle", label.y.npc = "top" )# Add equation#+ ylim(0, 25) #+ xlim(0,


ggplot(all.deet, aes(Genome.Length..bp.,sum, fill=phage_type, colour = phage_type)) +
  geom_point() +
  geom_smooth()


spor <- subset(all.deet, all.deet$f_spor==TRUE)
nospor <- subset(all.deet, all.deet$f_spor==FALSE)

ggplot(spor, aes(Genome.Length..bp.,sum, fill=phage_type, colour = phage_type)) +
  geom_point() +
   geom_smooth(method="lm")+ # Add regression line
 stat_regline_equation(aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~~")),
                        label.x.npc = "middle", label.y.npc = "top" )# Add equation#+ ylim(0, 25) #+ xlim(0,


ggplot(nospor, aes(Genome.Length..bp.,sum, fill=phage_type, colour = phage_type)) +
  geom_point() +
   geom_smooth(method="lm")+ # Add regression line
 stat_regline_equation(aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~~")),
                        label.x.npc = "middle", label.y.npc = "top" )# Add equation#+ ylim(0, 25) #+ xlim(0,




```